app/ui/src/app/customizations/api-connector/api-connector.reducer.ts

```diff
@@ -215,9 +216,17 @@ export function apiConnectorReducer(
           warnings: action.payload.warnings,
           errors: action.payload.errors,
           name: action.payload.name ? action.payload.name
-                                    : action.configuredProperties.specification.info.title,
+                                    : action.configuredProperties
+                                      && action.configuredProperties.specification
+                                      && action.configuredProperties.specification.info
+                                      && action.configuredProperties.specification.info.title
```

Comment:
![](https://i.redd.it/y60p260597x11.jpg)

(nothing to suggest here, just maybe that we _really_ need a better model definition?)

Reply:
Actually that bit can be refactored, not sure why we dig into the specification there, there are properties set on the backend, let me see if that works...

Reply:
Yeah, backend guarantees to return those properties even if they're not specified, in that case the value is `"unspecified"`.